autoscraper/auto_scraper.py

```diff
@@ -17,37 +19,34 @@ def __init__(self, stack_list=None, url=None):
 
     def save(self, file_path):
         with open(file_path, 'w') as f:
-            f.write(json.dumps(self.stack_list))
+            json.dump(self.stack_list, f)
 
     def load(self, file_path):
         with open(file_path, 'r') as f:
-            self.stack_list = json.loads(f.read())
+            self.stack_list = json.load(f)
 
-    @staticmethod
-    def _get_soup(url=None, html=None, request_args=None):
+    @classmethod
+    def _get_soup(cls, url=None, html=None, request_args={}):
         if html:
             return BeautifulSoup(html, 'lxml')
-        request_args = request_args if request_args else {}
-        headers = dict(AutoScraper.request_headers)
+
+        headers = dict(cls.request_headers)
         if url:
             headers['Host'] = urlparse(url).netloc
+
         headers = request_args.get('headers', headers)
         html = requests.get(url, headers=headers, **request_args).text
+
         return BeautifulSoup(html, 'lxml')
 
-    @staticmethod
-    def _get_valid_attrs(item):
-        attrs = dict(item.attrs)
-        for attr in item.attrs:
-            if attr not in {'class', 'style'}:
-                del attrs[attr]
-                continue
-            if attrs[attr] == []:
-                attrs[attr] = ''
-        return attrs
+    @classmethod
+    def _get_valid_attrs(cls, item):
+        return {
+            k: v or '' for k, v in item.attrs if k in {'class', 'style'}
+        }
```

Comment:
It changes all False values (like `0`, `False`) to '', which is something we don't want.

Reply:
This is not fixed yet.

Reply:
weird... my repo was okay and it was synced with remote... pushed new changes, seems fixed with new push.

Reply:
It has two problems:
- We should check for empty list not `None`. 
- Iterating on `item.attrs` returns only the key. 

Anyway it's hard to add rules to this style as you see, I prefer the original one. More readable and easier to maintain and add new rules.